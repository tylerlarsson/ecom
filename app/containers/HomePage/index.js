/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import Button from '@material-ui/core/Button';
import routes from 'constants/routes.json';

export function HomePage({ history }) {
  return (
    <div style={{ textAlign: 'center' }}>
      <h1>Ecom Freedom Homepage test</h1>
      <Button variant="contained" color="primary" onClick={() => history.push(routes.LOGIN)}>
        Login
      </Button>
    </div>
  );
}

HomePage.propTypes = {
  history: PropTypes.object
};

const mapStateToProps = createStructuredSelector({});

const withConnect = connect(mapStateToProps);

export default compose(
  withConnect,
  withRouter,
  memo
)(HomePage);
