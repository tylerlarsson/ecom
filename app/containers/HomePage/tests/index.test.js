import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';

import history from 'utils/history';
import HomePage from '../index';

const mockStore = configureMockStore();
const store = mockStore({});

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(
      <Provider store={store}>
        <BrowserRouter history={history}>
          <HomePage />
        </BrowserRouter>
      </Provider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
