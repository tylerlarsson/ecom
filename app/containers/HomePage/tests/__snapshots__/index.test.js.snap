// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  style="text-align: center;"
>
  <h1>
    Ecom Freedom Homepage
  </h1>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Login
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
